gapis/resolve/memory.go

```diff
@@ -339,3 +320,73 @@ func MemoryAsType(ctx context.Context, p *path.MemoryAsType, rc *path.ResolveCon
 
 	return vals[0], nil
 }
+
+// Function memoryAsType resolves typed memory related raw data,
+// and return the decoded memory.
+// This is a private resolving function inside gapis,.
+func memoryAsType(ctx context.Context, s *api.GlobalState, rng memory.Range, poolId uint32, typeIndex uint64,
+	p path.Node, rc *path.ResolveConfig) (*memory_box.Value, error) {
+	// Check whether the requested pool was ever created.
+	pool, err := s.Memory.Get(memory.PoolID(poolId))
+	if err != nil {
+		return nil, &service.ErrDataUnavailable{Reason: messages.ErrInvalidMemoryPool(poolId)}
+	}
+	sz := rng.Size
+	if sz == 0 {
+		sz = 0xFFFFFFFFFFFFFFFF
+	}
+
+	dec := s.MemoryDecoder(ctx, pool.Slice(memory.Range{
+		Base: rng.Base,
+		Size: sz,
+	}))
+
+	e, err := types.GetType(typeIndex)
```

Comment:
Can we change to longer names? It's not clear why the return of GetType is named "e". What does "e" stand for?

Reply:
Fixed in new commit: Address comment #3.